buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java

```diff
@@ -1,287 +0,0 @@
-                "firebaseLibrary", FirebaseLibraryExtension.class, project, LibraryType.ANDROID);
-
-    LibraryExtension android = project.getExtensions().getByType(LibraryExtension.class);
-
-    android.compileOptions(
-        options -> {
-          options.setSourceCompatibility(JavaVersion.VERSION_1_8);
-          options.setTargetCompatibility(JavaVersion.VERSION_1_8);
-        });
-
-    // In the case of and android library signing config only affects instrumentation test APK.
-    // We need it signed with default debug credentials in order for FTL to accept the APK.
-    android.buildTypes(
-        (NamedDomainObjectContainer<BuildType> types) ->
-            types
-                .getByName("release")
-                .setSigningConfig(types.getByName("debug").getSigningConfig()));
-    android.defaultConfig(
-        cfg -> {
-          cfg.buildConfigField("String", "VERSION_NAME", "\"" + project.getVersion() + "\"");
-        });
-
-    // see https://github.com/robolectric/robolectric/issues/5456
-    android.testOptions(
-        (TestOptions options) ->
-            options
-                .getUnitTests()
-                .all(
-                    closureOf(
-                        test -> {
-                          test.systemProperty("robolectric.dependency.repo.id", "central");
-                          test.systemProperty(
-                              "robolectric.dependency.repo.url", "https://repo1.maven.org/maven2");
-                          test.systemProperty("javax.net.ssl.trustStoreType", "JKS");
-                        })));
-
-    // skip debug tests in CI
-    // TODO(vkryachko): provide ability for teams to control this if needed
-    if (System.getenv().containsKey("FIREBASE_CI")) {
-      android.setTestBuildType("release");
-      project
-          .getTasks()
-          .all(
-              task -> {
-                if ("testDebugUnitTest".equals(task.getName())) {
-                  task.setEnabled(false);
-                }
-              });
-    }
-
-    setupApiInformationAnalysis(project, android);
-    android.testServer(new FirebaseTestServer(project, firebaseLibrary.testLab, android));
-
-    setupStaticAnalysis(project, firebaseLibrary);
-
-    configurePublishing(project, firebaseLibrary, android);
-
-    // reduce the likelihood of kotlin module files colliding.
-    project
-        .getTasks()
-        .withType(
-            KotlinCompile.class,
-            kotlin ->
-                kotlin
-                    .getKotlinOptions()
-                    .setFreeCompilerArgs(
-                        ImmutableList.of("-module-name", kotlinModuleName(project))));
-
-    project.getPluginManager().apply(DackkaPlugin.class);
-    project.getPluginManager().apply(GitSubmodulePlugin.class);
-    project.getTasks().getByName("preBuild").dependsOn("updateGitSubmodules");
-  }
-
-  private static void setupApiInformationAnalysis(Project project, LibraryExtension android) {
```

Comment:
This should deff be refactored into a `Task`. No need to do that in this PR though, just a note for later.